src/Validation/Constraints/Parallel.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/* Copyright (c) 2017 Stefan Hecken <stefan.hecken@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\Validation\Constraints;
+use ILIAS\Validation\Constraint;
+use ILIAS\Data\Factory;
+use ILIAS\Data\Result;
+
+class Parallel extends Custom implements Constraint {
+	/**
+	 * @var Constraint[]
+	 */
+	protected $constraints;
+
+	/**
+	 * @var Constraint[]
+	 */
+	protected $failed_constraints = array();
```

Comment:
This is bad, since it makes the Constraint stateful. There is no other way than to recalculate the failed constraints in `getErrorMessage`, at least I can't see one.